apps/govern/common-util/functions/resetState.ts

```diff
@@ -0,0 +1,23 @@
+import { ActionCreatorWithoutPayload } from '@reduxjs/toolkit';
+import { queryClient } from 'context/Web3ModalProvider';
+import { clearState } from 'store/govern';
+import { AppDispatch } from 'store/index';
+
+// Reset previously saved data so it's re-fetched automatically
+export const resetState = (
+  keys: string[],
+  dispatch: AppDispatch,
+  clearAction: ActionCreatorWithoutPayload<
+    'govern/clearState' | 'govern/clearUserState'
+  > = clearState,
+) => {
+  queryClient.removeQueries({
+    predicate: (query) =>
+      keys.includes(
+        // `query.queryKey[0]` is just the name of the function that initiate the query cache, e.g "readContract"
```

Comment:
@mohandast52 I've made it shared and explained what are queryKey[0]/[1], to make it clear as you asked about it in previous PR